compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -1058,7 +1058,7 @@ class TypeComparer(initctx: Context) extends ConstraintHandling[AbsentContext] w
     }
 
     // begin recur
-    if (tp2 eq NoType) false
+    if (tp1.eq(NoType) || tp2.eq(NoType)) false
```

Comment:
This code should be as short as possible, so that it's inlined. Let's not add another test here unless we must do so.

In this case it turns out that adding the clause

          case NoType => false

at line 552 in compareRec works as well. So I prefer that.

Reply:
Moved it to line 552